### DISCUSSION TITLE
Homebrew reinstall on Apple Silicon, reinstalling after official support

### DISCUSSION THREAD

**Discussion by johnalanwoods** [Homebrew/discussions#624] [Everyday usage] [answered]
Is homebrew "officially" supported on Apple Silicon at this point?

If I originally installed brew back when it was in beta status, do I need to reinstall or is the version updated throughout development on my current machine OK?

**Comment by tjluoma** [comment]
1. Not officially supported yet. I suspect it will be widely reported when it happens. Also, I have seen other messages that have said there will be a way to gracefully handle the switchover from Intel to M1 when the time comes, so I'm not going to worry about it until it is ready (and then I'll probably still wait a week or so).

2. I'm no expert, and I'm not clear what was in beta when you installed brew, but…having said that… given that `brew` updates itself, I would _think_ that it would be OK. Running `brew update` and then `brew upgrade` followed by `brew doctor` should tell you if there are any issues.

**Comment by SMillerDev** [comment, marked as answer]
> Is homebrew "officially" supported on Apple Silicon at this point?

Brew will run fine, but almost all software is beta quality.

> If I originally installed brew back when it was in beta status, do I need to reinstall or is the version updated throughout development on my current machine OK?

Brew update will handle everything for you.
